### PR TITLE
Adds SwiftLint

### PR DESCRIPTION
**Pull request by ashfurrow**
[SwiftLint](https://github.com/realm/SwiftLint) available via Homebrew:

```sh
brew install SwiftLint
```

There are two `.swiftlint.yml` files, one for our app target and one for our test target. I think this is important because we don't ship tests, so I don't mind force-unwrapping optionals there. All of these changes are open to discussion of course.

I decided not to include it on CI for the same reason we don't use spacecommander there. I looked into only linting changed files, but that introduced problems integrating with Xcode. Linting takes 0.67 seconds on my machine, so it's something to keep an eye on if it gets too lengthy (we could add the ability to specify more than one path at once to `swiftlint`, which would fix our Xcode integration problem).

### CODE REVIEW

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
        
            2 Errors
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">No CHANGELOG changes made</td>
    </tr>
        <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">There were <a href="https://eigen-ci.s3.amazonaws.com/2748/index.html">snapshot errors</a></td>
    </tr>
    
    
  </tbody>
</table>
  

  
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            2 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |252.5ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-swiftlint/Artsy/View_Controllers/Auction/AuctionViewController.swift#L181) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
207.4ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-swiftlint/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
174.1ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-swiftlint/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
80.0ms | [Observable.swift](/artsy/eigen/tree/ashfurrow-swiftlint/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
71.5ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/ashfurrow-swiftlint/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
70.7ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/ashfurrow-swiftlint/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, sideMargin: String)
62.3ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/ashfurrow-swiftlint/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L63) | func stackView() -> ORStackView




<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

File: Artsy.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -5182,6 +5184,34 @@
 			shellScript = "if type -p clang-compilation-database-tool > /dev/null 2>&1; then\n  clang-compilation-database-tool collect \"${OBJECT_FILE_DIR_normal}\" > \"${SRCROOT}/compiler_commands.json\"\nfi";
 			showEnvVarsInLog = 0;
 		};
+		5E38181E1D0338DE00FA56C0 /* Swift Lint */ = {
```
Comment (orta) [inline Artsy.xcodeproj/project.pbxproj]: hrm, FWIW I won't be installing swiftlint while this is here, build times are already bad enough adding a second doesn't fly from my perspective

Comment (orta) [inline Artsy.xcodeproj/project.pbxproj]: Probably the best way to get me on board with that is a similar approach to spacecommander, a verify script on the commit

**Comment by ashfurrow** [comment]
Getting intermittent error on CI, it's consistently "got expected a matching snapshot in looks_good_when_stuck_to_the_bottom", example: https://eigen-ci.s3.amazonaws.com/2748/index.html I'll take a look later.

File: Artsy.xcodeproj/project.pbxproj (hunk 1 above)
Comment (ashfurrow) [inline Artsy.xcodeproj/project.pbxproj]: Okay, I'll see what I can do. A precommit hook won't highlight problems inline with Xcode, but that's something I could use the Xcode plugin for on my own machine.

Comment (orta) [inline Artsy.xcodeproj/project.pbxproj]: I'm fine with that, but you're gonna see a lot of my warnings then 👯

Comment (orta) [inline Artsy.xcodeproj/project.pbxproj]: Looks like this was created for pretty much this exact use-case: https://github.com/norio-nomura/ExtraBuildPhase

Comment (ashfurrow) [inline Artsy.xcodeproj/project.pbxproj]: Not quite sure what that does?

Comment (ashfurrow) [inline Artsy.xcodeproj/project.pbxproj]: Ah, I see. So like, per-user linting. Neat.

**Comment by ashfurrow** [comment]
Ha, test failure was actually legit: https://github.com/artsy/eigen/pull/1612/commits/973ac57d3a7d78a24f93aaa35d12c01797bc9ea3

After removing the build scripts, I added SwiftLint invocations to a precommit hook (installed via a ruby script from `make artsy`). We can optimize later for how things go using the precommit hook, maybe we want to restrict to only files changed, but I think this is a reasonable foundation.

**Comment by ashfurrow** [comment]
Build is only failing due to `Developer Specific file shouldn't be changed` in Danger, unit tests are passing.

File: Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift [hunk 2]
```diff
@@ -6,7 +6,7 @@ import Then
 import MARKRangeSlider
 import ARAnalytics
 
-private let CellIdentifier = "Cell"
+private let cellIdentifier = "Cell"
```
Comment (orta) [inline Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift]: This one I'm not too convinced on, I guess the logic is that it's a "variable" of something, even if that something is the global scope

Comment (orta) [inline Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift]: not a blocker though

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionBidHistoryViewController.swift [hunk 3]
```diff
@@ -109,7 +109,7 @@ class LiveAuctionBidHistoryViewController: UITableViewController {
     }
 
     override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        return tableView.dequeueReusableCellWithIdentifier("live")!
+        return tableView.dequeueReusableCellWithIdentifier("live", forIndexPath: indexPath)
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionBidHistoryViewController.swift]: weird, didn't know about this

**Comment by orta** [comment]
Yep 👍

**Comment by orta** [comment]
Note: Danger can ignore specific warnings/fails, I think you add this to the body of the PR:

`Danger Ignore: Developer Specific file shouldn't be changed`

**Comment by ashfurrow** [comment]
Cool!

**Pull request merged**